contracts/utils/TimestampedHashRegistry.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./interfaces/ITimestampedHashRegistry.sol";
+
+// TODO: would SelfMulticall be useful?
+contract TimestampedHashRegistry is Ownable, EIP712, ITimestampedHashRegistry {
+    using ECDSA for bytes32;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    mapping(bytes32 => EnumerableSet.AddressSet) private _hashTypeToSigners;
+
+    mapping(bytes32 => SignedHash) public hashTypeToSignedHash;
+
+    bytes32 private constant _SIGNED_HASH_TYPE_HASH =
+        keccak256(
+            "SignedHash(bytes32 hashType,bytes32 hash,uint256 timestamp)"
+        );
+
+    constructor() EIP712("TimestampedHashRegistry", "1.0.0") {}
+
+    function setHashTypeSigners(
+        bytes32 hashType,
+        address[] calldata signers
+    ) external onlyOwner {
+        require(hashType != bytes32(0), "Hash is zero");
+        require(signers.length != 0, "Signers length is empty");
+        for (uint256 ind = 0; ind < signers.length; ind++) {
+            _hashTypeToSigners[hashType].add(signers[ind]);
+        }
+        emit SetHashTypeSigners(hashType, signers);
+    }
+
+    function registerSignedHash(
+        bytes32 hashType,
+        SignedHash calldata signedHash,
```

Comment:
I don't see why the struct is not spread (though if you remove the struct altogether based on the comment above this is no longer a concern)

Reply:
May I ask why is this a concern? Is the issue that I'm using a `struct` type as argument and this might be more expensive that just having 2 arguments? I saw that this was a common pattern like in [here](https://github.com/api3dao/airnode-protocol-v1/blob/102-TimestampedHashRegistry/contracts/utils/ExpiringMetaTxForwarder.sol#L77) and [here](https://github.com/api3dao/omnimarket/blob/main/contracts/Omnimarket.sol#L268)

Reply:
The first example is trying to be similar to the OpenZeppelin example meta tx forwarder contract (which no longer exists), the second example is doing it to avoid stack size limit errors. Otherwise I avoid it if I can, because for example it's not clear if/how the manager multisig frontend supports this.